core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/stacks/request/tags/upgrade/UpgradeV4Request.java

```diff
@@ -15,23 +15,26 @@
 @ValidUpgradeRequest
 public class UpgradeV4Request {
 
-    @ApiModelProperty(ModelDescriptions.UpgradeModelDescription.IMAGE_ID)
+    @ApiModelProperty(UpgradeModelDescription.IMAGE_ID)
     private String imageId;
 
-    @ApiModelProperty(ModelDescriptions.UpgradeModelDescription.RUNTIME)
+    @ApiModelProperty(UpgradeModelDescription.RUNTIME)
     private String runtime;
 
-    @ApiModelProperty(ModelDescriptions.UpgradeModelDescription.LOCK_COMPONENTS)
+    @ApiModelProperty(UpgradeModelDescription.LOCK_COMPONENTS)
     private Boolean lockComponents;
 
-    @ApiModelProperty(ModelDescriptions.UpgradeModelDescription.DRY_RUN)
+    @ApiModelProperty(UpgradeModelDescription.DRY_RUN)
     private Boolean dryRun;
 
     private Boolean replaceVms = Boolean.TRUE;
 
-    @ApiModelProperty(ModelDescriptions.UpgradeModelDescription.SHOW_AVAILABLE_IMAGES)
+    @ApiModelProperty(UpgradeModelDescription.SHOW_AVAILABLE_IMAGES)
     private UpgradeShowAvailableImages showAvailableImages = UpgradeShowAvailableImages.DO_NOT_SHOW;
 
+    @ApiModelProperty(hidden = true)
+    private boolean internal;
```

Comment:
could you replace this with a class which makes it posible to extend for later unforeseen scenarios? boolean is too restrictive and not extendable.
also I think this should be validated. If it's set and not called with internal crn then the request should be rejected